### ISSUE TITLE
Steam Guard does not get saved/cached

### ISSUE THREAD

**Issue by Claasclever** [dumbasPL/csgo-checker#30]
Everytime I try to load the current status of my accounts the white box appears prompting me to enter my steam guard code. I already tried multiple releases and made sure the C:\Users\<USER>\AppData\Local\Programs\csgo-checker folder was empty after uninstalling.  I remeber you didn't had to enter your steam guard everytime you try to load the cooldowns. Am I doing something wrong on my end or is this an already known issue?

**Comment by dumbasPL** [comment]
It will only remember the email-based steam guard. If you have the mobile steam guard you will need to input the shared secret for it in the account details so the checker can generate codes automatically. I'm not sure why it doesn't remember the mobile steam guard because I've never looked into it. In theory it should be possible since the steam client does it just fine.

Also, the checker stores information in AppData\Roaming and not AppData\Local

There is another big issue that is slowly creeping up and that is the fact that we aren't using the new login system that steam added (the one with the qr code). Once we switch to that then it should work because we will have a permanent refresh token.

So TLDR for now: if you want to use mobile auth with this program then you will need the shared secret for it.

**Comment by Claasclever** [comment]
I was referring to the email-based steam guard. It still does happen, even after clearing all the data from AppData\Roaming after uninstalling. If there is any way to generate any logs i would be happy to share them with you

**Comment by dumbasPL** [comment]
Logs won't really help if you manage to log in successfully after inputting the code.

Can you confirm a few things for me:

- Are you using any sort of VPN/proxy?
- Can you successfully save the password in the Steam client?
- When you get the Steam guard email, does it say that it's a login from a new location?
- Is this happening with all accounts or just specific ones?

Can you try the following:

1. close the program
2. delete the `%localappdata%\doctormckay\node-steamuser` directory
3. open the checker and verify that it still doesn't remember the steam guard
4. go to the above-mentioned directory and verify that you do/don't have a `sentry.<random id>.bin` file

**Comment by Claasclever** [comment]
- I'm not using a VPN.
- Yes, I can save the the password in the Steam client
- Yes, it does say "It looks like you are trying to log in from a new device."
- It is happening with all my accounts, that have an Email Guard.

I followed your steps and deleted the mentioned directory. After verifying that it still doesn't remember my steam guard, two file got created.

- cellid-c28c912d1b4f9ec38bfc3e1a15973524.txt
- cm_list.json

**Comment by dumbasPL** [comment]
Hmm, interesting. So it seems like the sentry file is not being created at all. This file stores the saved session and allows you to log in again without a password/steam guard. Now we need to figure out why.

**Comment by dumbasPL** [comment]
Ok, so it seems like Steam has finally disabled the sentry files and an update will be needed.  
Relevant update in steam-user: https://github.com/DoctorMcKay/node-steam-user/releases/tag/v4.29.0

Feel free to bump this issue if it's not fixed in a few days in case I forget ;)

**Comment by dumbasPL** [comment]
Update released, everything seems to work as expected now. Please try it out and report back.

https://github.com/dumbasPL/csgo-checker/releases/tag/v1.4.3

**Comment by Claasclever** [comment]
Works with both E-Mail Steam Guard code and Mobile code. Thx for the quick fix.
